src/lib/spinel/radio_spinel.hpp

```diff
@@ -34,30 +34,16 @@
 #ifndef RADIO_SPINEL_HPP_
 #define RADIO_SPINEL_HPP_
 
-#include "openthread-posix-config.h"
-
 #include <openthread/platform/radio.h>
 
-#if OPENTHREAD_POSIX_CONFIG_RCP_UART_ENABLE
-#include "hdlc_interface.hpp"
-#endif
-
-#if OPENTHREAD_POSIX_CONFIG_RCP_SPI_ENABLE
-#include "spi_interface.hpp"
-#endif
-
-#if !OPENTHREAD_POSIX_CONFIG_RCP_UART_ENABLE && !OPENTHREAD_POSIX_CONFIG_RCP_SPI_ENABLE
-#error "Please enable either OPENTHREAD_POSIX_CONFIG_RCP_UART_ENABLE or OPENTHREAD_POSIX_CONFIG_RCP_SPI_ENABLE."
-#endif
-
+#include "spinel.h"
 #include "spinel_interface.hpp"
-#include "lib/spinel/spinel.h"
 #include "ncp/ncp_config.h"
 
 namespace ot {
-namespace Posix {
+namespace Spinel {
 
-class RadioSpinel : public SpinelInterface::Callbacks
+template <typename InterfaceType, typename ProcessContextType> class RadioSpinel
```

Comment:
suggestion: add documentation explaining the template types (and any requirements/restrictions they may have, e.g., `InterfaceType` should sub-class so and so and/or provide so and so methods, etc). Thanks

Reply:
Comment added.